README.md

```diff
@@ -1,2 +1,93 @@
-# oop-library
-School library App
+# OOP school library
+
+> In this project, I will start building my school library app. I'm practicing design patterns for building OOP applications.
+## Built With
+
+- Ruby
+
+## Getting Started
+
+To get a local copy up and running follow these simple example steps.
+
+### Prerequisites
+
+- Git/GitHub;
+- Optional ( IDE installed in your machine ) - recommended IDE: Vscode;
+- Ruby Installed in your machine;
+
+### Setup
+
+- To clone this repo:
+  - Open the Terminal(Mac/Linux) or the Cmd (Windows);
+  - Navigate to where you want to paste the Repo folder;
+
+## install 
+
+
+To install and run this project on your local machine, follow these steps:
+
+1. Clone the repository to your local machine using Git:
+
+   ```shell
+   git clone https://github.com/AlyAtalla/oop-library.git
+
+
+## usage 
+
+# Example Ruby code or usage:
+
+# Usage example:
+your_instance = YourClass.new
+result = your_instance.some_method
+puts result
+
+
+## Key Features 
+Describe between 1-3 key features of the application.
```

Comment:
- [ ] Please kindly remove this text as it is only there to help you customise the section properly :+1: